framework/transport/ieee1905_transport/test/ieee1905_transport_test.cpp

```diff
@@ -251,18 +251,10 @@ int main(int argc, char *argv[])
     while ((c = getopt(argc, argv, "b:i:u:x")) != -1) {
         switch (c) {
         case 'b':
-            ieee1905_bridge_if_index = if_nametoindex(optarg);
-            if (ieee1905_bridge_if_index == 0) {
-                fprintf(stderr, "unknown interface %s\n", optarg);
-                return false;
-            }
+            ieee1905_bridge_ifname = std::string(optarg, IF_NAMESIZE);
```

Comment:
This will do a serious out-of-bounds access, because it copies exactly IF_NAMESIZE characters even if optarg is shorter!
```suggestion
            ieee1905_bridge_ifname = std::string(optarg);
```